src/test/kotlin/com/atlassian/performance/tools/report/RootWorkspaceTest.kt

```diff
@@ -0,0 +1,30 @@
+package com.atlassian.performance.tools.report
+
+import com.atlassian.performance.tools.workspace.api.RootWorkspace
+import org.assertj.core.util.Files
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+
+class RootWorkspaceTest {
+
+    @Test
+    fun shouldCreatingNewInstanceHaveNoSideEffect() {
+        // given
+        val tempDir = Files.newTemporaryFolder()
+        // when
+        RootWorkspace(parent = tempDir.toPath())
+        // then
+        assertThat(tempDir).isEmptyDirectory()
+    }
+
+    @Test
+    fun shouldCreateDirWhenGetCurrentTask() {
+        // given
+        val tempDir = Files.newTemporaryFolder()
+        // when
+        RootWorkspace(parent = tempDir.toPath()).currentTask
```

Comment:
You can use `java.nio.file.Files.createTempDirectory` and have it as a `Path` from the get-go.

Reply:
I know and I picked assertj one as it's easier to use and read the code